pyvista/core/filters/image_data.py

```diff
@@ -2287,3 +2287,31 @@ def _validate_dimensional_operation(
             )
 
         return dimensions_mask, dimensions_result
+
+    def magnify(
+        self, *, factor: VectorLike[int] = (1, 1, 1), progress_bar: bool = False
```

Comment:
Can also allow single `int` inputs and enable the `broadcast` option in `validate_array3`.
```suggestion
        self, *, factor: int | VectorLike[int] = (1, 1, 1), progress_bar: bool = False
```